RecoParticleFlow/PFClusterProducer/plugins/SimMappers/RealisticHitToClusterAssociator.h

```diff
@@ -125,9 +120,12 @@ class RealisticHitToClusterAssociator
                 }
                 else
                 {
+                    partialEnergies.resize(numberOfClusters,0.f);
+                    unsigned int layer = layerId_[hitId];
+                    float sumE = 0.f;
                     for(unsigned int clId = 0; clId < numberOfClusters; ++clId )
                     {
-
+                        float energyDecayLength = getDecayLength(layer, fhOffset, bhOffset);
```

Comment:
Take out of the loop, otherwise it will be uselessly recomputed at each iteration